Godot: Add attachment handling documentation (#14210)
getsentry/sentry-docs: docs/platforms/godot/enriching-events/attachments/index.mdx

---
title: Attachments
description: "Learn more about how Sentry can store additional files in the same request as event attachments."
---

Sentry can enrich your events for further investigation by storing additional files, such as config or log files, as attachments.

## Creating Attachments

> The simplest way to create an attachment is to use a `path`. The SDK will read the contents of the file each time it prepares an event or transaction, then adds the attachment to the same [envelope](https://develop.sentry.dev/sdk/data-model/envelopes/). If the SDK can't read the file, the SDK logs an error message and drops the attachment.

Author: This is a link to the developer docs – I don't think it would work as internal link. Is there a better way to link to dev docs?

> The simplest way to create an attachment is to use a `path`. The SDK will read the contents of the file each time it prepares an event or transaction, then adds the attachment to the same [envelope](https://develop.sentry.dev/sdk/data-model/envelopes/). If the SDK can't read the file, the SDK logs an error message and drops the attachment.

Reviewer: Oops, my bad. I missed that it switched over to develop when I first clicked into it. We can leave it as is.

</Alert>

In addition, you can set these parameters:

`filename`

The filename is the name of the file to display in Sentry. When using bytes you have to specify a filename, whereas with a path you don't as the SDK is going to use the last path component.

`content_type`

The specific media content type that determines how the attachment is rendered in the Sentry UI. Any [MIME type](https://www.iana.org/assignments/media-types/media-types.xhtml) may be used; the default is `application/octet-stream`.

 We currently support and can render the following MIME types:

- `text/plain`
- `text/css`
- `text/csv`
- `text/html`
- `text/javascript`
- `text/json` or `text/x-json` or `application/json` or `application/ld+json`
- `image/jpeg`
- `image/png`
- `image/gif`

## Uploading Attachments

You can add attachments that will be sent with every event using <PlatformIdentifier name="SentrySDK.add-attachment()" /> method.

<PlatformContent includePath="enriching-events/attachment-upload" />

<Alert>

> Sentry allows at most 20MB for a compressed request, and at most 100MB of uncompressed attachments per event, including the crash report file (if applicable). Uploads exceeding this size are rejected with HTTP error `413 Payload Too Large` and the data is dropped immediately. To add larger or more files, consider secondary storage options.

Reviewer: "To add larger or more files, consider secondary storage options."

Do we mean that we'll accept this data if integrated with a secondary storage platform? Or just that Sentry does not accept files larger than 20mb/100mb at all?

Author: We can follow up on this. This block is also present in Unity and Unreal, so its scope is larger than this PR’s.

</Alert>

> Attachments persist for 30 days; if your total storage included in your quota is exceeded, attachments will not be stored. You can delete attachments or their containing events at any time. Deleting an attachment does not affect your quota - Sentry counts an attachment toward your quota as soon as it is stored.

Reviewer: ```suggestion
Attachments are saved for 30 days. Newly submitted attachments will not be stored If your total storage quota has been exceeded. You can delete attachments or their containing events at any time. Deleting an attachment does not affect your quota - Sentry counts an attachment toward your quota as soon as it is stored.
```
